docs/userguide/campaigns.md

```diff
@@ -1,25 +1,125 @@
 # Campaigns
 
```

Comment:
this userguide for my taste is maybe even a bit too close to simply being an example, at teh same time the campaign is prob the most 

As I imagine suer guides, they do not focus on a sepcific uxe case, as in this example here, but they enumerate all related functionality with brief explanations and put them in a grand context of the entire package (something not done eg in doctrings). There are quite many aspects I feel are missing:
- explain whats the batch size
- explain the Adrianian mantra of recommendations of different size are not necessarily the same (joint optimization)
- explain that we cache recommendations if they are requested repeatedly
- data addition should only happen via `add_measurements` as baybe also has some metadata
- this is the place to talk about serializability of the entire object and why this is cool and that its also dangerous and that it also has limits
- this is also the place to explain that a campaign can be created from a config

Reply:
I decided to start working on this already in this PR. Current state (will update this comment):

- Included information on what the batch size is and the Adrianian mantra

Reply:
I think we are moving in the right direction, but there are still several things that I feel need to be adjusted. Currently trying to figure out what is the fastest way forward. Perhaps it's fastest if we just add our changes commits on top?

Reply:
i thought we do this just for typos etc ??

Reply:
Ok, I just have the feeling that too many comments from too many people here are overwhelming for such a discussion. Perhaps I'll first let you two resolve your current open issues before I start flagging stuff in parallel....

Reply:
@Scienfitz @AdrianSosic : As far as I see it, there is no progress happening here. So if you are happy with the **general structure** of this user guide, please add a comment to **this** thread such that we can close it and open smaller threads as the ones already opened by @Scienfitz.

Reply:
fine for me as I had already a "round" on this PR, did not resolve because Adrian is outstanding, I seem to remember he wanted to wait with his review until after my round